dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java

```diff
@@ -0,0 +1,321 @@
+        {
+            throw new BadRequestException( "Program stage is specified but does not exist: " + programStageUid );
+        }
+
+        return programStage;
+    }
+
+    private OrganisationUnit validateOrgUnit( String orgUnitUid )
+        throws BadRequestException
+    {
+        if ( orgUnitUid == null )
+        {
+            return null;
+        }
+
+        OrganisationUnit orgUnit = organisationUnitService.getOrganisationUnit( orgUnitUid );
+        if ( orgUnit == null )
+        {
+            throw new BadRequestException( "Org unit is specified but does not exist: " + orgUnitUid );
+        }
+
+        return orgUnit;
+    }
+
+    private void validateUser( User user, Program pr, ProgramStage ps, String orgUnitUid, OrganisationUnit orgUnit )
+        throws ForbiddenException
+    {
+        if ( pr != null && !user.isSuper() && !aclService.canDataRead( user, pr ) )
+        {
+            throw new ForbiddenException( "User has no access to program: " + pr.getUid() );
+        }
+
+        if ( ps != null && !user.isSuper() && !aclService.canDataRead( user, ps ) )
+        {
+            throw new ForbiddenException( "User has no access to program stage: " + ps.getUid() );
+        }
+
+        if ( orgUnitUid != null && !trackerAccessManager.canAccess( user, pr, orgUnit ) )
+        {
+            throw new ForbiddenException( "User does not have access to orgUnit: " + orgUnit );
+        }
+    }
+
+    private TrackedEntity validateTrackedEntity( String trackedEntityUid )
+        throws BadRequestException
+    {
+        if ( trackedEntityUid == null )
+        {
+            return null;
+        }
+
+        TrackedEntity trackedEntity = trackedEntityService.getTrackedEntity( trackedEntityUid );
+        if ( trackedEntity == null )
+        {
+            throw new BadRequestException( "Tracked entity is specified but does not exist: " + trackedEntityUid );
+        }
+
+        return trackedEntity;
+    }
+
+    private void validateAttributeOptionCombo( CategoryOptionCombo attributeOptionCombo, User user )
+        throws ForbiddenException
+    {
+        if ( attributeOptionCombo != null && !user.isSuper()
+            && !aclService.canDataRead( user, attributeOptionCombo ) )
+        {
+            throw new ForbiddenException(
+                "User has no access to attribute category option combo: " + attributeOptionCombo.getUid() );
+        }
+    }
+
+    private void validateOperationParams( EventOperationParams params, User user, Program program )
+        throws BadRequestException
+    {
+        if ( params.getOrgUnitSelectionMode() != null )
+        {
+            String violation = getOuModeViolation( params, user, program );
+            if ( violation != null )
+            {
+                throw new BadRequestException( violation );
+            }
+        }
+    }
+
+    private String getOuModeViolation( EventOperationParams params, User user, Program program )
```

Comment:
```suggestion
    private String getOrgUnitModeViolation( EventOperationParams params, User user, Program program )
```

we should consistently use `orgUnit` everywhere 😄